### ISSUE TITLE
Add labels to node information in the application pod view.

### ISSUE THREAD

**Issue by msoderberg** [argoproj/argo-cd#15274]
# Summary

Add labels to node information in the application pod view.

# Motivation

We're providing multi-tenant clusters with limited access to the Kubernetes API, and users often use ArgoCD to get detailed information about their deployed applications. For our users to get a better understanding of where their pods end up, it would be useful to present more information about the nodes in the UI.

Example labels:
* `topology.kubernetes.io/zone`
* `karpenter.sh/provisioner-name`
* `karpenter.sh/capacity-type`

# Proposal

Add node labels to `HostInfo` and find a good way of presenting them in the UI. Not all labels are useful, so some sort of filtering would be nice (eg. add a `node-pod-view-labels` configuration option to the application controller or something like that).

I've started working on this and would like to ask for some input before I spend more time on it. Is it a useful feature from your point of view? Any feedback on the proposed solution? Any ideas on how to present this in the UI would be greatly appreciated. 

Here's my WIP branch and a screenshot of how it looks at the moment: https://github.com/argoproj/argo-cd/compare/master...msoderberg:argo-cd:feat/node-labels-in-pod-view
![argocd-pod-view](https://github.com/argoproj/argo-cd/assets/5403544/18fe0aed-0232-4b47-b8cb-3430aa724a93)

**Comment by blakepettersson** [comment]
cc @rbreeze @keithchong

**Comment by agaudreault** [comment]
@msoderberg To configure this feature, the config pattern could either be

1. Per project
2. A flat list of allowed labels to display in the argocd-cm
3. A list of allowed labels to display in the argocd-cm, with a cluster glob filter (similar to `resource.exclusions`)

I do think that 2 is what is necessary. 3 and 1 seems to offer an unnecessary level of granularity.

**Comment by msoderberg** [comment]
Thanks for the feedback @agaudreault!

Option 2 would work for our use-case and I can update the PR accordingly if we all agree on that. 

Personally I'd prefer to configure this per project and add support for inheritance from global projects (currently missing in my PR). That way ArgoCD admins could be responsible for allowing labels that makes sense across all projects but still allow for adding custom labels on project level.

**Comment by agaudreault** [comment]
@msoderberg do you have a use case where this would require a per-project/per-cluster configuration? I cannot think of any scenario where a different set of labels would need to be displayed based on which project an App belongs to (1).

I also think that configuring it per cluster (3) is not really necessary, because if the configured labels are not present on the node, then they should just be ignored.

**Comment by msoderberg** [comment]
Sure! In our setup each team have their own project, and parts of the project configuration are managed by the teams themselves. If this configuration would be on project level, a team could decide to visualize labels that are useful for them, but that could be noise to others (e.g. specific Karpenter labels or custom labels that the owner of the node(s) have set).

It might be an unlikely use-case that is not worth the complexity compared to having a list of labels in the argocd-cm.

I completely agree about option 3 :+1:

**Comment by agaudreault** [comment]
Feel free to open a PR for option 3 :)

**Comment by blakepettersson** [comment]
I think option 1 is fine - at least if we get that done we could extend that to option 2 or 3 in the future if need be

**Comment by agaudreault** [comment]
> I think option 1 is fine - at least if we get that done we could extend that to option 2 or 3 in the future if need be

I disagree. Option 1 requires adding a new Alpha feature and modifies the AppProject resource that might not be necessary from the start. Option 2 and 3 are controller config with a lot less impact if we modify the feature later on.

**Comment by blakepettersson** [comment]
@agaudreault sorry I miswrote - I meant option 2! (that's what I get for reading this late at night). 

So if we have a simple list, that sounds good to me. If it turns out that we need globbing or more granular project-level permissions we can add that at a later time.

**Comment by msoderberg** [comment]
I think we all agree that option 2 is the way to go for now 👍 I decided to close my old PR and create a new one: #23260 Please take a look and let me know what you think.
